cvat/apps/onepanelio/views.py

```diff
@@ -39,3 +95,248 @@ def get_workflow_templates(request):
+
+
+    with tempfile.TemporaryDirectory() as test_dir:
+
+        project.export(dump_format, test_dir, save_images=True)
+
+        if cloud_provider == "s3":
+            
+            import boto3
+            from botocore.exceptions import ClientError
+
+            #check if datasets folder exists on aws bucket
+            s3_client = boto3.client('s3')
+
+            try:
+                s3_client.head_object(Bucket=bucket_name, Key=cloud_prefix)
+                #add logging
+            except ClientError:
+                # Not found
+                slogger.glob.info("Datasets folder does not exist in the bucket, creating a new one.")
+                s3_client.put_object(Bucket=bucket_name, Key=(cloud_prefix))
+
+
+            for root,dirs,files in os.walk(test_dir):
+                for file in files:
+                    upload_dir = root.replace(test_dir, "")
+                    if upload_dir.startswith("/"):
+                        upload_dir = upload_dir[1:]
+                    s3_client.upload_file(os.path.join(root,file),bucket_name,os.path.join(cloud_prefix,dataset_name, upload_dir, file))
+          
+        elif cloud_provider == "gcs":
+            from google.cloud import storage
+            storage_client = storage.Client()
+            bucket = storage_client.bucket(bucket_name)
+
+            for root, dirs, files in os.walk(test_dir):
+                for file in files:
+                    upload_dir = root.replace(test_dir, "")
+                    if upload_dir.startswith("/"):
+                        upload_dir = upload_dir[1:]
+                    blob = bucket.blob(os.path.join(cloud_prefix, dataset_name, upload_dir, stamp))
+                    blob.upload_from_filename(os.path.join(root, file))
+        
+        elif cloud_provider == "az":
+            pass
+
+        else:
+            raise ValueError("Invalid cloud provider! Should be from ['s3','gcs','az']")
+
+    return bucket_name, dataset_name
+
+
+@api_view(['POST'])
+def create_annotation_model(request, pk):
+    """
+        Executes workflow selected by User.
+    """
+    
+    db_task = TaskModel.objects.get(pk=pk)
+    db_labels = db_task.label_set.prefetch_related('attributespec_set').all()
+    db_labels = {db_label.id:db_label.name for db_label in db_labels}
+    num_classes = len(db_labels.values())
+
+    slogger.glob.info("Creating annotation model for task: {} with num_classes {}".format(db_task.name,num_classes))
+
+    form_data = json.loads(request.body.decode('utf-8'))
+    slogger.glob.info("Form data without preprocessing {} {}".format(form_data, type(form_data)))
+ 
+    form_args = form_data['arguments']
+    time = datetime.now()
+    stamp = time.strftime("%m%d%Y%H%M%S")
+
+    if "cpu" in form_data['machine_type']:
+        tf_image = "tensorflow/tensorflow:1.13.1-py3"
+        machine ="Standard_D4s_v3"
```

Comment:
I noticed we're hard-coded the machine type here? Are we unable to do this dynamically at this point?